azure/services/roleassignments/roleassignments.go

```diff
@@ -103,6 +103,9 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 	for _, roleAssignmentSpec := range s.Scope.RoleAssignmentSpecs(principalID) {
 		log.V(2).Info("Creating role assignment")
+		if roleAssignmentSpec.ResourceName() == "" {
+			log.V(2).Info("RoleAssignmentName is empty. This is not expected and will cause this System Assigned Identity to have no permissions.")
```

Comment:
The webhook already add a UUID when the role assignment name is empty if the System Assigned identity is set.

In my case, my mistake, was that because of an object filter I added to the webhook but I did not add as a label to the machindeployment spec.template.spec.metadata my machines were not hitting the webhook ending up with an empty role assignment name.

A more clear logging line would have helped me a lot :)

But I understand is a very edge case so I can just drop this in case

Reply:
Can we increase the verbosity to V(4) if we plan on merging this?

Reply:
- At first glance, the log line sounds to me more like a "bug" which should have been solved rather than a verbose debug output. Is there something we could improve upon? Probably add validations at webhook level or improve the defaulting functions?
- Else, do you think we could add this to the documentation to let the user know instead of adding a log line?

Reply:
@nawazkh the PR got merged shortly after your comment but can totally be reverted if you feel it needs more attention :pray: 

>  the log line sounds to me more like a "bug" which should have been solved rather than a verbose debug output. Is there something we could improve upon? Probably add validations at webhook level or improve the defaulting functions?

the way i hit this issue was _because due to some mistakes on my hand with `objectSelector` and labels_ some resources, in particular the azureMachines created by a MachineDeployment, were not hitting the webhook ( which already handles this situation by adding a random UUID as `roleAssignmentName` if one is not set but required )

So i am not sure how to handle this other than at `controller time` when the `roleAssignmentName` is used. 

* should we error rather than log ? 

> Else, do you think we could add this to the documentation to let the user know instead of adding a log line?

Yes i guess ?

My logic for adding a log line was something like this 

* I made a mistake on my side ( so i am not sure how to handle that at the docs level, how many other type of issues could cause this ? ) 
* We hit a situation that the controller could not really handle well ( the controller thought that the `empty named roleassignment` was already reconciled on the system assigned identity )
* The information from the logs were not really helpful to understand i was actually missing that field ( due to issues completely outside of the controller scope ) 

I would have saved probably quite a few hours of work if i got a log line like this :)

Reply:
@primeroz Thank you for adding more context! I am happy that this log line will help you going ahead, I don't think we need to revert it.